src/charm.py

```diff
@@ -179,6 +182,29 @@ def _on_get_initial_password_action(self, event):
             logger.error("Action get-initial-password failed. Replica consensus not exists")
             event.fail("Default admin password has not been generated yet.")
 
+    def _on_rotate_wordpress_secrets_action(self, event):
+        """Handle the rotate-wordpress-secrets action."""
+        if self._replica_consensus_reached():
```

Comment:
There is a way to refactor this to reduce nesting and improve readability, pseudo code below. It reduces the context that the reader has to maintain in their mind as they read the code. Do you think it would improve the code?

```Python
if not self._replica_consensus_reached():
    <log and fail the action>

if not self.unit.is_leader():
    <log and fail the action>

<code for action execution>
```

Reply:
Thanks for this suggestion.
Updated.